### PR TITLE
Set maxBuffer for docker loading strategy via configuration value

### PR DESCRIPTION
**Pull request by coeniebeyers**
Proposal for https://github.com/trufflesuite/truffle/issues/3380
Thanks to @rynobey for also debugging/fixing this on our end.

### CODE REVIEW

**Comment by eggplantzzz** [comment]
Hey @coeniebeyers, thanks for the PR! So far Truffle doesn't have any precedent for using environment variables, is there any way you could make this something that you would configure? Check out the current configuration options [here](https://www.trufflesuite.com/docs/truffle/reference/configuration#compiler-configuration). Perhaps it could be listed under the `compilers` property? Something like `compilers.solc.spawn.maxBuffer`? What do you think?

**Comment by coeniebeyers** [comment]
Hi @eggplantzzz! np! Ah, yes, then using something like `compilers.solc.spawn.maxBuffer` makes sense. 

Please take a look at the updated PR. Is there somewhere I can add this to documentation?

Please note that I'm trying to test by locally installing truffle from source in our project with `npm install <path to truffle>` but even after linking with lerna it doesn't seem to be able to do that, so there might be some things wrong still.

**Comment by coeniebeyers** [comment]
@eggplantzzz thanks for the info, I was able to test locally and it looks good after my latest commit. Please let me know if anything needs to change.

**Comment by eggplantzzz** [comment]
One way that I test is that I clone the repo, make my changes, and then run `packages/core/cli.js <command>`. Maybe you will find that helpful. I'll take a look at your changes, thanks!

File: packages/compile-solidity/compilerSupplier/loadingStrategies/Docker.js [hunk 1]
```diff
@@ -6,12 +6,15 @@ const semver = require("semver");
...
 
-// Set a sensible limit for maxBuffer
-// See https://github.com/nodejs/node/pull/23027
-const maxBuffer = 1024 * 1024 * 100;
-
 class Docker extends LoadingStrategy {
   async load() {
+    // Set a sensible limit for maxBuffer
+    // See https://github.com/nodejs/node/pull/23027
+    let maxBuffer = 1024 * 1024 * 100;
```
Comment (eggplantzzz) [inline packages/compile-solidity/compilerSupplier/loadingStrategies/Docker.js:13]: Another option here is
```
const maxBuffer = this.config.spawn && this.config.spawn.maxBuffer ?
  this.config.spawn.maxBuffer :
  1024 * 1024 * 100;
```
but what you did is good too!

**Review by eggplantzzz — APPROVED** [review]
I think this looks good to me.

**Review by fainashalts — APPROVED** [review]
LGTM as well! Thanks @coeniebeyers!

**Comment by coeniebeyers** [comment]
Sure, np!

**Pull request merged**